swarm/api/api.go

```diff
@@ -361,3 +363,50 @@ func (self *Api) BuildDirectoryTree(mhash string, nameresolver bool) (key storag
 	}
 	return key, manifestEntryMap, nil
 }
+
+// Look up mutable resource updates at specific periods and versions
+func (self *Api) DbLookup(key storage.Key, name string, period uint32, version uint32) (storage.Key, io.ReadSeeker, int, error) {
+	var err error
+	if version != 0 {
+		if period == 0 {
+			currentblocknumber, err := self.resource.GetBlock()
+			if err != nil {
+				return nil, nil, 0, fmt.Errorf("Could not determine latest block: %v", err)
+			}
+			period = self.resource.BlockToPeriod(name, currentblocknumber)
+		}
+		_, err = self.resource.LookupVersionByName(name, period, version, true)
+	} else if period != 0 {
+		_, err = self.resource.LookupHistoricalByName(name, period, true)
+	} else {
+		_, err = self.resource.LookupLatestByName(name, true)
+	}
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	key, data, err := self.resource.GetContent(name)
+	if err != nil {
+		return nil, nil, 0, err
+	}
+	return key, bytes.NewReader(data), len(data), nil
```

Comment:
Given this method always loads the full data, should it not just return `[]byte` rather than `bytes.Reader` + `int`?